### ISSUE TITLE
400,401 status codes of oauth2 response are not properly parsed, throws TypeError: cannot create weak reference to 'NoneType' object

### ISSUE THREAD

**Issue by ashleylst** [OpenIDC/pyoidc#807]
Basically, I did 
``` 
resp = client.do_any(
          Message,
          request_args=body,
          endpoint=endpoint,
          body_type="json",
          method="POST",
          authn_method="client_secret_basic",
)
```

and I am supposed to get a 400 error response, but instead, an exception is thrown, traceback is as follows:

>  File "/usr/lib/python2.7/site-packages/oic/oauth2/\_\_init\_\_.py", line 784, in do_any
    state=state, http_args=http_args)
>  File "/usr/lib/python2.7/site-packages/oic/oauth2/\_\_init\_\_.py", line 660, in request_and_return
    **kwargs)
 > File "/usr/lib/python2.7/site-packages/oic/oauth2/\_\_init\_\_.py", line 601, in parse_request_response
    if issubclass(response, ErrorResponse):
 > File "/usr/lib64/python2.7/abc.py", line 180, in \_\_subclasscheck\_\_
    if issubclass(subclass, scls):
 > File "/usr/lib64/python2.7/abc.py", line 180, in \_\_subclasscheck\_\_
    if issubclass(subclass, scls):
 > File "/usr/lib64/python2.7/abc.py", line 184, in \_\_subclasscheck\_\_
    cls._abc_negative_cache.add(subclass)
 > File "/usr/lib64/python2.7/_weakrefset.py", line 84, in add
   > self.data.add(ref(item, self._remove))
>TypeError: cannot create weak reference to 'NoneType' object

As suggested, the code that leads to this exception is here: https://github.com/OpenIDC/pyoidc/blame/0d316fc0baca21b16164ef17fb275979d178f4b6/src/oic/oauth2/__init__.py#L740.
And this line of code doesn't make sense in the context, perhaps it was left unintentionally. I am not seeing why there is a `elif` case for 400,401 errors and does nothing but lead to an exception.

**Comment by schlenk** [comment]
Indeed, that check looks useless.

The code might have tried to allow passing in a custom ErrorResponse class as response parameter, but failed to do it properly.

**Comment by infohash** [comment]
@schlenk What should be the expected behaviour? Can I return status code there?

**Comment by schlenk** [comment]
Should be fixed with #817.
